File: src/main/java/chat/teco/tecochat/member/domain/MemberRepository.java
package chat.teco.tecochat.member.domain;

import static chat.teco.tecochat.member.exception.MemberExceptionType.NOT_FOUND_MEMBER;

import chat.teco.tecochat.member.exception.MemberException;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberRepository extends JpaRepository<Member, Long> {

    boolean existsByName(String name);

    Optional<Member> findByName(String name);

    default Member getByName(String name) {
        return findByName(name)
                .orElseThrow(() -> new MemberException(NOT_FOUND_MEMBER));
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/member/MemberRepository.kt
package chat.teco.tecochat.domain.member

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.findByIdOrNull

fun MemberRepository.getByIdOrThrow(id: Long) = findByIdOrNull(id)
    ?: throw NoSuchElementException("사용자가 존재하지 않습니다. id: $id")

interface MemberRepository : JpaRepository<Member, Long> {
    fun findByName(name: String): Member?
}
